snake/snake_game_khr.py

```diff
@@ -0,0 +1,174 @@
+    pass
+
+
+class Game:
+    def __init__(self, width, height):
+        self.symbol = "o"
+        self.width = width
+        self.height = height
+        self.board = Board(self.width, self.height)
+        self.snake = Snake()
+        self.apple = Apple()
+
+    def init_apple(self):
+        last_pos = self.apple.position
+        self.board = Board(self.width, self.height)
+        positions = []
+        for x in range(self.height):
+            for y in range(self.width):
+                z = (item for item in self.snake.body)
+                if (x, y) != last_pos and (x, y) not in z:
+                    if self.board.board[x][y] == ' ':
+                        positions.append((x, y))
+
+        if positions:
+            position = random.choices(positions)[0]
+            self.apple = Apple(position=position, symbol="$")
+            return self.apple.position
+        else:
+            raise GameOverError('No places for apple!')
+
+    def play(self):
+        game_cycles = 100
+        try:
+            self.render()
+            while game_cycles > 0:
+                i, j = self.snake.body[-1]
+                possible_moves = self.snake.choices()
+                next_moves = []
+                for move in possible_moves:
+                    if move in self.snake.body:
+                        continue
+                    if move[0] != 0 and move[1] != 0 and move[0] != (self.height-1) and move[1] != (self.width-1):
+                        next_moves.append(move)
+                # print(next_moves)
+                distances = []
+                for move in next_moves:
+                    dist = self.board.distance(move, self.apple.position)
+                    distances.append(dist)
+                # print(distances)
+                res = dict(zip(next_moves, distances))
+                try:
+                    min(res, key=res.get)
+                except ValueError:
+                    print('Game over!')
+                    break
+                next_move = min(res, key=res.get)
+                if next_move == self.apple.position:
+                    self.snake.eat(next_move)
+                    self.render()
+                    self.init_apple()
+                    self.render()
+                    continue
+                self.snake.move(next_move)
+                self.render()
+                game_cycles -= 1
+        except GameOverError:
+            print('Game Over!')
+
+    def clear(self):
+        self.board = Board(self.width, self.height)
+        if os.name == 'posix':
+            os.system('clear')
+        else:
+            os.system('cls')
+
+    def render(self):
+        self.clear()
+        i, j = self.apple.position
+        self.board.board[i][j] = self.apple.symbol
+        for i, j in self.snake.body:
+            self.board.board[i][j] = self.snake.symbol
+        if (i, j) == self.snake.body[-1]:
+            self.board.board[i][j] = "@"
```

Comment:
I think you should not have (i, j) == self.snake.body[-1]. You render apple and snake separately always.

Reply:
> I think you should not have (i, j) == self.snake.body[-1]. You render apple and snake separately always.

But '@' here is a symbol of the snake's head, not apple (which is '$'), or what do you mean here?